### PR TITLE
Fix botattr validation and correct flag usage.

### PR DESCRIPTION
**Pull request by Cizzle**
Found by: Geo
Patch by: Cizzle
Fixes: #597 

One-line summary: Fixes validation of the flags that can be added and uses the correct flags as wanted for the `botattr` command.

### CODE REVIEW

**Comment by vanosg** [comment]
```
.botattr bar +s
[02:13:05] #-HQ# botattr bar +s
Bot flags for bar are now +s.
.botattr bar +z
[02:13:09] #-HQ# botattr bar +z
Bot flags for bar are now +s.
.botattr bar +0
[02:13:12] #-HQ# botattr bar +0
Bot flags for bar are now +s0.
.botattr bar +1
[02:13:15] #-HQ# botattr bar +1
Bot flags for bar are now +s01.
.botattr bar +9
[02:13:18] #-HQ# botattr bar +9
Bot flags for bar are now +s019.
```

LGTM

**Comment by vanosg** [comment]
I'm a little conused- 
As far as I can tell from the doc, ULL is C99. But, it does compile with CFLAGS="-std=c89 -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=500 -D_DEFAULT_SOURCE -Wall" ./configure

Any thoughts?

File: src/users.h [hunk 1]
```diff
@@ -97,7 +97,7 @@ struct user_entry {
     char *string;
     void *extra;
     struct list_type *list;
-    unsigned long ulong;
+    unsigned long long ulong;
```
Comment (thommey) [inline src/users.h]: can we just go with uint64_t?

**Comment by Cizzle** [comment]
ULL (and uint64_t) are POSIX2001. From the POSIX standard it wasn't clear to me though if uint64_t is present on every machine: "If an implementation provides integer types with width 64 that meet these requirements, then the following types are required: int64_t uint64_t".

**Comment by thommey** [comment]
we have stdint.h and an autoconf macro with a fallback that will declare uint64_t if there is any 64bit type available. I don't know what to do if they don't exist, I don't think that's possible on any system we support and would be more in line with the rest of the code (and C99 which has those types, if available)

**Comment by vanosg** [comment]
build

**Comment by michaelortmann** [comment]
Problem:

```
.+bot bar
.botattr bar +s019   
```                                                                                                                                                          

result:

32bit eggdrop: `Bot flags for bar are now +s01.`

64bit eggdrop: `Bot flags for bar are now +s019.`

that is what the compiler under 32 bit told us, only it should have thrown an error not a warning:

```
gcc -g -O2 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/usr/include/tcl8.6 -g3 -DDEBUG -DDEBUG_ASSERT -DDEBUG_MEM -DDEBUG_DNS  -c flags.c
flags.c: In function ‘set_user_flagrec’:
flags.c:656:35: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     set_user(&USERENTRY_BOTFL, u, (void *) fr->bot);
                                   ^
flags.c: In function ‘get_user_flagrec’:
flags.c:706:15: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
     fr->bot = (unsigned long long) get_user(&USERENTRY_BOTFL, u);
               ^
flags.c: In function ‘botfl_set’:
flags.c:778:29: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
   unsigned long long atr = ((unsigned long long) buf & BOT_VALID);
                             ^
flags.c: In function ‘botfl_tcl_set’:
flags.c:822:21: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     botfl_set(u, e, (void *) fr.bot);
```

**Comment by vanosg** [comment]
LGTM

```
.botattr bot1 +a
Bot flags for bot1 are now +a.
.botattr bot1 +u
[22:39:43] #-HQ# botattr bot1 +u
Bot flags for bot1 are now +a.
.botattr bot1 +v
[22:39:45] #-HQ# botattr bot1 +v
Bot flags for bot1 are now +a.
.botattr bot1 +0
[22:39:51] #-HQ# botattr bot1 +0
Bot flags for bot1 are now +a0.
.botattr bot1 +9
[22:39:53] #-HQ# botattr bot1 +9
Bot flags for bot1 are now +a09.
.botattr bot1 +1
[22:39:55] #-HQ# botattr bot1 +1
Bot flags for bot1 are now +a019.
.botattr bot1 +z
[22:42:50] #-HQ# botattr bot1 +z
Bot flags for bot1 are now +a019.
```

**Comment by vanosg** [comment]
To capture a LOOONG discussion on IRC on this- we can't introduce ULLs because it breaks 32-bit systems. To use all 0-9, it requires 36 bits on the mask. So, we decided to eliminate v-z from the bitmask register, so as not to affect anyone using custom 0-9 flags, and just not ever use the hard coded v-z flags, as we have plenty of open flags to use still. 

Or, if we ever do need to use them either a) we won't be supporting 32-bit by then or b) all of the current devs making the decision will be long departed from this project, and it is someone else's problem to fix :)

**Pull request merged**